direct/src/showbase/GarbageReport.py

```diff
@@ -212,7 +211,7 @@ def run(self):
                     startIndex = 0
                     # + 1 to include a reference back to the first object
                     endIndex = numObjs + 1
-                    if type(objs[-1]) is types.InstanceType and type(objs[0]) is dict:
+                    if hasattr(objs[-1], '__dict__') and type(objs[0]) is dict:
```

Comment:
I'm not at all certain about this replacement.  I guess the goal was to identify old-style vs. new-style classes.  So wouldn't the right thing to do just be to remove this `if` block altogether, since all classes are new-style classes?

I'm not sure what having a `__dict__` indicates.

Reply:
This is now removed in 3ac4f3a1988ee304f5c1fa415a0fd701a9826b04 as it is actually a no-op, thank you!

Reply:
It's not a no-op, it's meant to just reorder the list of classes in the cycle.  A very common occurrence in a cycle is "instance -> its dict", and if the cycle is reported with the break across this boundary it's just making sure it's still showing the instance first, since that makes the report a smidge easier to read.

I'll be restoring this line when I check in the fix.  Initially I thought it was about identifying old-style classes, but I think all classes should be handled this way.